### PR TITLE
Update etcd topology only after original request succeed

### PR DESCRIPTION
**Pull request by CyberDem0n**
There is no point to try to update topology until original request is
not performed. Also for us it is more important to execute original
request rather then keep topology of etcd cluster in sync.

In addition to that implement the same retry-timeout logic in the
`machines` property which already is used in `api_execute` method.

### CODE REVIEW

File: tests/test_etcd.py [hunk 1]
```diff
@@ -132,6 +132,10 @@ def socket_getaddrinfo(*args):
 def http_request(method, url, **kwargs):
     if url == 'http://localhost:2379/timeout':
         raise ReadTimeoutError(None, None, None)
+    if url == 'http://localhost:2379/v2/machines':
+        ret = MockResponse()
+        ret.content = 'http://http://localhost:2379,http://localhost:4001'
```
Comment (alexeyklyukin) [inline tests/test_etcd.py]: too many http://

**Comment by alexeyklyukin** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
👍

**Pull request merged**